src/zenml/integrations/gcp/service_connectors/gcp_service_connector.py

```diff
@@ -86,6 +86,28 @@ class GCPUserAccountCredentials(AuthenticationConfig):
         "distribute the user account credentials JSON to clients instead.",
     )
 
+    @root_validator(pre=True)
+    def validate_user_account_dict(
+        cls, values: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Convert the user account credentials to JSON if given in dict format.
+
+        Args:
+            values: The configuration values.
+
+        Returns:
+            The validated configuration values.
+
+        Raises:
+            ValueError: If the user account credentials JSON is not a valid
+                JSON object.
+        """
+        if isinstance(values.get("user_account_json"), dict):
```

Comment:
This is just something left over from the "use service connectors to authenticate secrets stores" PR and revealed during testing this PR.